extensions/amp-accordion/1.0/amp-accordion.js

```diff
@@ -98,20 +102,47 @@ function getState(element, mu) {
       CONTENT_SHIM_PROP,
       bindContentShimToElement
     );
-    const expanded = section.hasAttribute('expanded');
+    const expandStateShim = memo(
+      section,
+      EXPAND_STATE_SHIM_PROP,
+      getExpandStateTrigger
+    );
     const props = dict({
       'key': section,
       'as': sectionShim,
       'headerAs': headerShim,
       'contentAs': contentShim,
-      'expanded': expanded,
+      'expanded': section.hasAttribute('expanded'),
       'id': section.getAttribute('id'),
+      'onExpandStateChange': expandStateShim,
     });
     return <AccordionSection {...props} />;
   });
   return dict({'children': children});
 }
 
+/**
+ * @param {!Element} section
+ * @return {Function}
+ */
+function getExpandStateTrigger(section) {
+  const element = section.parentElement;
+  const action = Services.actionServiceForDoc(
+    /** @type {!Element|!ShadowRoot} */ (element)
+  );
+  const triggerEvent = (expanded) => {
+    const eventName = expanded ? 'expand' : 'collapse';
+    const event = createCustomEvent(
+      toWin(element.ownerDocument.defaultView),
+      `accordionSection.${eventName}`,
+      dict()
```

Comment:
Should have been more clear, this argument can be removed entirely, no?

Reply:
No problem, actually the `gulp check-types` does not like it when we exclude this third argument :/

Reply:
Oh hm, it looked like the signature for `createCustomEvent` allows the given `detail` to be `undefined`, but since the type checker doesn't like it then we should keep.